sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -205,6 +206,11 @@ def __init__(self, base_user_agent=None, **kwargs):  # pylint: disable=super-ini
             )
 
         if application_id:
+            if application_id.length > self._MAX_APPLICATION_ID_LENGTH:
+                raise ValueError("'applicationId' length cannot be greater than " + self._MAX_APPLICATION_ID_LENGTH)
+            if ' ' in application_id:
+                raise ValueError("'applicationId' cannot contain spaces.")
+
```

Comment:
I would recommend combining these so that the user gets one error message. Otherwise, if the application ID is more than 24 characters AND has spaces, they will get runtime errors twice, once for each error. See:
https://github.com/Azure/azure-sdk-for-ios/pull/1000